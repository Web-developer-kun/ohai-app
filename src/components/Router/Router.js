import React, { Suspense } from "react";
import "../../bootstrap-social.css";
const Signin = React.lazy(() => import("../Signin/Signin"));
const Register = React.lazy(() => import("../Register/Register"));
const Placeholder = React.lazy(() => import("../Placeholder/Placeholder"));

class Router extends React.Component {
  render() {
    const { route } = this.props;
    if (route === "signin") {
      return (
        <div style={{ width: "100%", height: "100%" }}>
          <Suspense fallback={<h1>Signin</h1>}>
            <Signin {...this.props} />
          </Suspense>
        </div>
      );
    } else if (route === "register") {
      return (
        <div style={{ width: "100%", height: "100%" }}>
          <Suspense fallback={<h1>Register</h1>}>
            <Register {...this.props} />
          </Suspense>
        </div>
      );
    } else if (route === "placeholder") {
      return (
        <div style={{ width: "100%", height: "100%" }}>
          <Suspense fallback={<h1>Register</h1>}>
            <Placeholder {...this.props} />
          </Suspense>
        </div>
      );
    }
  }
}

export default Router;
