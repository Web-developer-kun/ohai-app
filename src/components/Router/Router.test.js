import { shallow } from "enzyme";
import React from "react";
import Router from "./Router";

let wrapper;
beforeEach(() => {
  const mockProps = {
    email: "",
    setPass1: "",
    setPass2: "",
    formErrMsg: "",
    password: "",
    signInEmail: "",
    signInPassword: "",
    route: "signin",
    onEmailChange: jest.fn(),
    onSetPass1: jest.fn(),
    onSetPass2: jest.fn(),
    setFormErrMsg: jest.fn(),
    setPassword: jest.fn(),
    onSignInEmailChange: jest.fn(),
    onSignInPasswordChange: jest.fn(),
    changeRoute: jest.fn()
  };
  wrapper = shallow(<Router {...mockProps} />);
});
it("renders Router component", () => {
  expect(wrapper).toMatchSnapshot();
});
