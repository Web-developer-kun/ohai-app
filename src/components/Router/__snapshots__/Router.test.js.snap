// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Router component 1`] = `ShallowWrapper {}`;
